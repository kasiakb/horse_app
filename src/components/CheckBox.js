import React from 'react';

class CheckBox extends React.Component {
  checkBoxes(checkBoxInput) {
    return checkBoxInput.map((forage) => {
      return (
        <div key={forage}> 
          <label>
            <input 
              name = {forage}
              type='checkbox'
              checked={this.props.checked}
              onChange={this.props.onChange}
            />
            {forage}
          </label>
        </div>
      )
    })
  }

  render() {
    return (
      <div className='pasza'>
        <form>
          {this.checkBoxes(this.props.checkBoxInput)}
        </form>
      </div>
    );
  } 
}

export default CheckBox;