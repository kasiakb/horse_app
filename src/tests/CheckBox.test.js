import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CheckBox from '../components/CheckBox';

describe('CheckBox component renders the checkboxes correctly', () => {
  it('renders correctly', () => {
    const heckBoxInput=[
      "Owies", 
      "Jęczmień", 
      "Wysłodki buraczane", 
      "Drozdze pastewne",
      "Siano"]
    const onChange = jest.fn()
    const checked = jest.fn()
    
    
    const rendered = renderer.create(
      <CheckBox
      checkBoxInput={heckBoxInput}
      onChange={onChange}
      checked={checked}
    />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});