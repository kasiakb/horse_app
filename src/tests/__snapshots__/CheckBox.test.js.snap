// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox component renders the checkboxes correctly renders correctly 1`] = `
<div
  className="pasza"
>
  <form>
    <div>
      <label>
        <input
          checked={[MockFunction]}
          name="Owies"
          onChange={[MockFunction]}
          type="checkbox"
        />
        Owies
      </label>
    </div>
    <div>
      <label>
        <input
          checked={[MockFunction]}
          name="Jęczmień"
          onChange={[MockFunction]}
          type="checkbox"
        />
        Jęczmień
      </label>
    </div>
    <div>
      <label>
        <input
          checked={[MockFunction]}
          name="Wysłodki buraczane"
          onChange={[MockFunction]}
          type="checkbox"
        />
        Wysłodki buraczane
      </label>
    </div>
    <div>
      <label>
        <input
          checked={[MockFunction]}
          name="Drozdze pastewne"
          onChange={[MockFunction]}
          type="checkbox"
        />
        Drozdze pastewne
      </label>
    </div>
    <div>
      <label>
        <input
          checked={[MockFunction]}
          name="Siano"
          onChange={[MockFunction]}
          type="checkbox"
        />
        Siano
      </label>
    </div>
  </form>
</div>
`;
